pkg/osquery/extension.go

```diff
@@ -35,17 +35,20 @@ type settingsStoreWriter interface {
 // and servers -- It provides a grpc and jsonrpc interface for
 // osquery. It does not provide any tables.
 type Extension struct {
-	NodeKey             string
-	Opts                ExtensionOpts
-	registrationId      string
-	knapsack            types.Knapsack
-	serviceClient       service.KolideService
-	settingsWriter      settingsStoreWriter
-	enrollMutex         sync.Mutex
-	done                chan struct{}
-	interrupted         atomic.Bool
-	slogger             *slog.Logger
-	logPublicationState *logPublicationState
+	NodeKey                       string
+	Opts                          ExtensionOpts
+	registrationId                string
+	knapsack                      types.Knapsack
+	serviceClient                 service.KolideService
+	settingsWriter                settingsStoreWriter
+	enrollMutex                   sync.Mutex
+	done                          chan struct{}
+	interrupted                   atomic.Bool
+	slogger                       *slog.Logger
+	logPublicationState           *logPublicationState
+	lastRequestQueriesTimestamp   *atomic.Int64
+	distributedForwardingInterval *atomic.Int64 // how frequently to forward RequestQueries requests to the cloud, in seconds
+	forwardAllDistributedUntil    *atomic.Int64 // allows for accelerated distributed requests until given timestamp
```

Comment:
I bet this one is going to need a mutex. Oh, or maybe that's what atomic does. I wonder if a mutex is faster. Eh.

Reply:
Chatted offline -- seph looked this up and determined atomic is (probably) faster! I will revisit in the next PR if necessary.